src/platform_impl/linux/kms/event_loop.rs

```diff
@@ -0,0 +1,701 @@
+            mode,
+            name: mode.name().to_string_lossy().into_owned(),
+        }))
+    }
+
+    #[inline]
+    pub fn available_monitors(&self) -> VecDeque<super::MonitorHandle> {
+        if let Some(mode) = *MODE.lock() {
+            self.device
+                .resource_handles()
+                .unwrap()
+                .connectors()
+                .iter()
+                .map(|f| super::MonitorHandle {
+                    connector: self.device.get_connector(*f).unwrap(),
+                    mode,
+                    name: mode.name().to_string_lossy().into_owned(),
+                })
+                .collect()
+        } else {
+            VecDeque::new()
+        }
+    }
+
+    #[inline]
+    pub fn raw_display_handle(&self) -> RawDisplayHandle {
+        let mut dh = DrmDisplayHandle::empty();
+        dh.fd = self.device.as_raw_fd();
+        dh.into()
+    }
+}
+
+fn find_plane(
+    planes: PlaneResourceHandles,
+    res: ResourceHandles,
+    crtc: &crtc::Info,
+    drm: &Card,
+) -> plane::Handle {
+    let (p_better_planes, p_compatible_planes): (
+        // The primary planes available to us
+        Vec<plane::Handle>,
+        // Other, not-ideal planes that are however useable
+        Vec<plane::Handle>,
+    ) = planes
+        .planes()
+        .iter()
+        .filter(|&&plane| {
+            // Get the plane info from a handle
+            drm.get_plane(plane)
+                .map(|plane_info| {
+                    let compatible_crtcs = res.filter_crtcs(plane_info.possible_crtcs());
+                    // Makes sure that the plane can be used with the CRTC we selected earlier
+                    compatible_crtcs.contains(&crtc.handle())
+                })
+                .unwrap_or(false)
+        })
+        .partition(|&&plane| {
+            // Get the plane properties from a handle
+            if let Ok(props) = drm.get_properties(plane) {
+                let (ids, vals) = props.as_props_and_values();
+                for (&id, &val) in ids.iter().zip(vals.iter()) {
+                    if let Ok(info) = drm.get_property(id) {
+                        // Checks if the plane is a primary plane, and returns true if it is,
+                        // if not it returns false
+                        if info.name().to_str().map(|x| x == "type").unwrap_or(false) {
+                            return val == (PlaneType::Primary as u32).into();
+                        }
+                    }
+                }
+            }
+            false
+        });
+
+    // Get the first (best) plane we find, or the first compatibile plane
+    *p_better_planes.get(0).unwrap_or(&p_compatible_planes[0])
+}
+
+fn find_card_path(seat_name: &str) -> Result<PathBuf, error::OsError> {
+    let mut enumerator = Enumerator::new().map_err(|e| {
+        os_error!(OsError::KmsError(format!(
+            "failed to open udev enumerator: {}",
+            e
+        )))
+    })?;
```

Comment:
Might be nice to use a combinator [like we have in `softbuffer`](https://github.com/rust-windowing/softbuffer/blob/ecfae194cebf97ca23e450e4f4541d911fa132a0/src/error.rs#L43-L60) to avoid duplicating this `map_err` block`.